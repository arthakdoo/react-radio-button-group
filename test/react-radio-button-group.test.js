import React from 'react';
import ReactRadioButtonGroup from '../src/react-radio-button-group';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import {mount} from 'enzyme';
import sinon from 'sinon';

test("Three text options yield three labels", () => {
    expect(
        renderer.create(
            <ReactRadioButtonGroup
                name='media'
                options={["TV", "Radio", "Youtube"]}/>
        ).toJSON()
    ).toMatchSnapshot();
});

test("Value provided means checked input", () => {
    expect(
        renderer.create(
            <ReactRadioButtonGroup
                name='media'
                options={["TV", "Radio", "Youtube"]}
                value="Radio"/>
        ).toJSON()
    ).toMatchSnapshot();
});

test("CSS classes all placed accordingly", () => {
    expect(
        renderer.create(
            <ReactRadioButtonGroup
                name='media'
                options={["TV", "Radio", "Youtube"]}
                value="Radio"
                groupClassName='cssGroup' itemClassName='cssItem'
                labelClassName='cssLabel' inputClassName='cssInput'/>
        ).toJSON()
    ).toMatchSnapshot();
});

test("CSS in option is stronger than general CSS", () => {
    expect(
        renderer.create(
            <ReactRadioButtonGroup
                name='media'
                options={
                [
                    {
                        value: "TV",
                        itemClassName: "cssItemTV",
                        labelClassName: "cssLabelTV"
                    },
                    {
                        value: "Youtube",
                        inputClassName: "cssInputYoutube"
                    },
                    "Radio"
                ]}
                value="Radio"
                groupClassName='cssGroup' itemClassName='cssItem'
                labelClassName='cssLabel' inputClassName='cssInput'/>
        ).toJSON()
    ).toMatchSnapshot();
});

test("Label is used when present, not value", () => {
    expect(
        renderer.create(
            <ReactRadioButtonGroup
                name='media'
                options={[
                    { value: "TV",      label: "Television" },
                    { value: "Youtube", label: "YouTube" },
                    { value: "Radio",   label: "Radio Station" }
                ]}/>
        ).toJSON()
    ).toMatchSnapshot();
});

test("fireOnMount set to true causes onChange when mounted", () => {
    const spy = sinon.spy();
    const group = mount(
        <ReactRadioButtonGroup
            name='media'
            options={["One", "Two", "Three"]}
            fireOnMount={true}
            isStateful={true}
            onChange={spy}
        />
    );
    expect(spy.callCount).toEqual(1);
});

test("fireOnMount not fired when stateless mode", () => {
    const spy = sinon.spy();
    const group = mount(
        <ReactRadioButtonGroup
            name='media'
            options={["One", "Two", "Three"]}
            fireOnMount={true}
            onChange={spy}
        />
    );
    expect(spy.callCount).toEqual(0);
});

test("onChange works correctly when radio clicked", () => {
    const spy = sinon.spy();
    const group = mount(
        <ReactRadioButtonGroup
            name='media'
            options={["One", "Two", {value: "Three", inputClassName: "cssThree"}]}
            isStateful={true}
            onChange={spy}
        />
    );
    expect(spy.callCount).toEqual(0);
    group.find('.cssThree').simulate('change');
    expect(spy.callCount).toEqual(1);
    expect(spy.firstCall.args.length).toEqual(1);
    expect(spy.firstCall.args[0]).toEqual('Three');
});


test("Clicking on an input in stateless mode has no independent effect", () => {
    const spy = sinon.spy();
    const group = mount(
        <ReactRadioButtonGroup
            name='media'
            options={[
                { value: "One", inputClassName: "cssOne" },
                "Two",
                { value: "Three", inputClassName: "cssThree" }
            ]}
            value="One"
            onChange={spy}
        />
    );
    group.find('.cssThree').simulate('change');
    expect(spy.callCount).toEqual(1);
    expect(spy.firstCall.args.length).toEqual(1);
    expect(spy.firstCall.args[0]).toEqual('Three');

    expect(group.find('.cssOne').  props().checked).toBe(true);
    expect(group.find('.cssThree').props().checked).toBe(false);
});

test("Clicking on an input in stateful mode has immediate effect", () => {
    const spy = sinon.spy();
    const group = mount(
        <ReactRadioButtonGroup
            name='media'
            options={[
                { value: "One", inputClassName: "cssOne" },
                "Two",
                { value: "Three", inputClassName: "cssThree" }
            ]}
            value="One"
            isStateful={true}
            onChange={spy}
        />
    );
    expect(group.find('.cssOne').props().checked).toBe(true);
    group.find('.cssThree').simulate('change');
    expect(spy.callCount).toEqual(1);
    expect(spy.firstCall.args.length).toEqual(1);
    expect(spy.firstCall.args[0]).toEqual('Three');

    expect(group.find('.cssOne').  first().props().checked).toBe(false);
    expect(group.find('.cssThree').first().props().checked).toBe(true);
});

test("Passing invalid onChange function does not crash and means no callback", () => {
    const group = mount(
        <ReactRadioButtonGroup
            name='media'
            options={["One", "Two", "Three"]}
            value="One"
            onChange={null}
        />
    );
    group.find('input').first().simulate('change');
    expect(2 + 2).toEqual(4);
});

test("Test disabled prop works", () => {
    expect(
        renderer.create(
            <ReactRadioButtonGroup
                name='media'
                options={["TV", "Radio", "Youtube"]}
                disabled={true}/>
        ).toJSON()
    ).toMatchSnapshot();
});
