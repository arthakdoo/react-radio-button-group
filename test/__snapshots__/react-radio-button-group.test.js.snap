exports[`test CSS classes all placed accordingly 1`] = `
<div
  className="cssGroup">
  <div
    className="cssItem">
    <input
      checked={false}
      className="cssInput"
      disabled={false}
      id="react-radio-button-group-3-TV"
      name="media"
      onChange={[Function]}
      type="radio"
      value="TV" />
    <label
      className="cssLabel"
      htmlFor="react-radio-button-group-3-TV">
      TV
    </label>
  </div>
  <div
    className="cssItem">
    <input
      checked={true}
      className="cssInput"
      disabled={false}
      id="react-radio-button-group-3-Radio"
      name="media"
      onChange={[Function]}
      type="radio"
      value="Radio" />
    <label
      className="cssLabel"
      htmlFor="react-radio-button-group-3-Radio">
      Radio
    </label>
  </div>
  <div
    className="cssItem">
    <input
      checked={false}
      className="cssInput"
      disabled={false}
      id="react-radio-button-group-3-Youtube"
      name="media"
      onChange={[Function]}
      type="radio"
      value="Youtube" />
    <label
      className="cssLabel"
      htmlFor="react-radio-button-group-3-Youtube">
      Youtube
    </label>
  </div>
</div>
`;

exports[`test CSS in option is stronger than general CSS 1`] = `
<div
  className="cssGroup">
  <div
    className="cssItemTV">
    <input
      checked={false}
      className="cssInput"
      disabled={false}
      id="react-radio-button-group-4-TV"
      name="media"
      onChange={[Function]}
      type="radio"
      value="TV" />
    <label
      className="cssLabelTV"
      htmlFor="react-radio-button-group-4-TV">
      TV
    </label>
  </div>
  <div
    className="cssItem">
    <input
      checked={false}
      className="cssInputYoutube"
      disabled={false}
      id="react-radio-button-group-4-Youtube"
      name="media"
      onChange={[Function]}
      type="radio"
      value="Youtube" />
    <label
      className="cssLabel"
      htmlFor="react-radio-button-group-4-Youtube">
      Youtube
    </label>
  </div>
  <div
    className="cssItem">
    <input
      checked={true}
      className="cssInput"
      disabled={false}
      id="react-radio-button-group-4-Radio"
      name="media"
      onChange={[Function]}
      type="radio"
      value="Radio" />
    <label
      className="cssLabel"
      htmlFor="react-radio-button-group-4-Radio">
      Radio
    </label>
  </div>
</div>
`;

exports[`test Label is used when present, not value 1`] = `
<div>
  <div>
    <input
      checked={false}
      disabled={false}
      id="react-radio-button-group-5-TV"
      name="media"
      onChange={[Function]}
      type="radio"
      value="TV" />
    <label
      htmlFor="react-radio-button-group-5-TV">
      Television
    </label>
  </div>
  <div>
    <input
      checked={false}
      disabled={false}
      id="react-radio-button-group-5-Youtube"
      name="media"
      onChange={[Function]}
      type="radio"
      value="Youtube" />
    <label
      htmlFor="react-radio-button-group-5-Youtube">
      YouTube
    </label>
  </div>
  <div>
    <input
      checked={false}
      disabled={false}
      id="react-radio-button-group-5-Radio"
      name="media"
      onChange={[Function]}
      type="radio"
      value="Radio" />
    <label
      htmlFor="react-radio-button-group-5-Radio">
      Radio Station
    </label>
  </div>
</div>
`;

exports[`test Test disabled prop works 1`] = `
<div>
  <div>
    <input
      checked={false}
      disabled={false}
      id="react-radio-button-group-12-TV"
      name="media"
      onChange={[Function]}
      type="radio"
      value="TV" />
    <label
      htmlFor="react-radio-button-group-12-TV">
      TV
    </label>
  </div>
  <div>
    <input
      checked={false}
      disabled={false}
      id="react-radio-button-group-12-Radio"
      name="media"
      onChange={[Function]}
      type="radio"
      value="Radio" />
    <label
      htmlFor="react-radio-button-group-12-Radio">
      Radio
    </label>
  </div>
  <div>
    <input
      checked={false}
      disabled={false}
      id="react-radio-button-group-12-Youtube"
      name="media"
      onChange={[Function]}
      type="radio"
      value="Youtube" />
    <label
      htmlFor="react-radio-button-group-12-Youtube">
      Youtube
    </label>
  </div>
</div>
`;

exports[`test Three text options yield three labels 1`] = `
<div>
  <div>
    <input
      checked={false}
      disabled={false}
      id="react-radio-button-group-1-TV"
      name="media"
      onChange={[Function]}
      type="radio"
      value="TV" />
    <label
      htmlFor="react-radio-button-group-1-TV">
      TV
    </label>
  </div>
  <div>
    <input
      checked={false}
      disabled={false}
      id="react-radio-button-group-1-Radio"
      name="media"
      onChange={[Function]}
      type="radio"
      value="Radio" />
    <label
      htmlFor="react-radio-button-group-1-Radio">
      Radio
    </label>
  </div>
  <div>
    <input
      checked={false}
      disabled={false}
      id="react-radio-button-group-1-Youtube"
      name="media"
      onChange={[Function]}
      type="radio"
      value="Youtube" />
    <label
      htmlFor="react-radio-button-group-1-Youtube">
      Youtube
    </label>
  </div>
</div>
`;

exports[`test Value provided means checked input 1`] = `
<div>
  <div>
    <input
      checked={false}
      disabled={false}
      id="react-radio-button-group-2-TV"
      name="media"
      onChange={[Function]}
      type="radio"
      value="TV" />
    <label
      htmlFor="react-radio-button-group-2-TV">
      TV
    </label>
  </div>
  <div>
    <input
      checked={true}
      disabled={false}
      id="react-radio-button-group-2-Radio"
      name="media"
      onChange={[Function]}
      type="radio"
      value="Radio" />
    <label
      htmlFor="react-radio-button-group-2-Radio">
      Radio
    </label>
  </div>
  <div>
    <input
      checked={false}
      disabled={false}
      id="react-radio-button-group-2-Youtube"
      name="media"
      onChange={[Function]}
      type="radio"
      value="Youtube" />
    <label
      htmlFor="react-radio-button-group-2-Youtube">
      Youtube
    </label>
  </div>
</div>
`;
